### DISCUSSION TITLE
Partial ArchiveDayData after PacketID mismatch

### DISCUSSION THREAD

**Discussion by tannegarn** [SBFspot/SBFspot#698] [General]
Irregularly, I do have partial archivedaydata from a Sunny Boy 5.0. The argument -D5 shows the following data:
```
Current Inverter Time: 16/05/2024 05:52:59
Inverter Wake-Up Time: 15/05/2024 21:19:46
Inverter Sleep Time  : 16/05/2024 05:52:42
********************
* ArchiveDayData() *
********************
startTime: 16/05/2024 `00:00:00`
--------: 00 01 02 03 04 05 06 07 08 09
00000000: 53 4D 41 00 00 04 02 A0 00 00 
00000010: 00 01 00 26 00 10 60 65 09 E0 
00000020: 99 01 25 A2 54 B3 00 00 7D 00 
00000030: DB B6 71 32 00 00 00 00 00 00 
00000040: 13 80 00 02 00 70 34 2F 45 66 
00000050: B4 80 46 66 00 00 00 00 
58 Bytes sent to [192.168.2.7]
58 bytes received from [192.168.2.7]
--------: 00 01 02 03 04 05 06 07 08 09
00000000: 53 4D 41 00 00 04 02 A0 00 00 
00000010: 00 01 00 26 00 10 60 65 09 A0 
00000020: FF FF FF FF FF FF 00 00 99 01 
00000030: 25 A2 54 B3 00 00 00 00 00 00 
00000040: F5 9F 00 02 80 51 00 48 21 00 
00000050: FF 4A 41 00 00 00 00 00 
<<<====== Content of pcktBuf =======>>>
--------: 00 01 02 03 04 05 06 07 08 09
00000000: 00 00 10 60 65 09 A0 FF FF FF 
00000010: FF FF FF 00 00 99 01 25 A2 54 
00000020: B3 00 00 00 00 00 00 F5 9F 00 
00000030: 02 80 51 00 48 21 00 FF 4A 41 
00000040: 00 00 00 00 
<<<=================================>>>
Packet ID mismatch. Expected 19, received 8181
logoffMultigateDevices()
logoffSMAInverter()
--------: 00 01 02 03 04 05 06 07 08 09
00000000: 53 4D 41 00 00 04 02 A0 00 00 
00000010: 00 01 00 22 00 10 60 65 08 A0 
00000020: FF FF FF FF FF FF 00 03 7D 00 
00000030: DB B6 71 32 00 03 00 00 00 00 
00000040: 14 80 0E 01 FD FF FF FF FF FF 
00000050: 00 00 00 00 
54 Bytes sent to [192.168.2.7]
```
Is the above a known issue? Is there any possibility to retry the get packet function when invalid data is received?

**Comment by sbf-** [comment]
What version of SBFspot is used?
Are there other SMA devices in your setup (e.g. Home Manager)?

**Reply by tannegarn** [reply]
[pvdiary_autorun_20240516.log](https://github.com/SBFspot/SBFspot/files/15334595/pvdiary_autorun_20240516.log)
SBFspot V3.9.8 is being used. The Sunny Boy 5.0 inverter is the only SMA device.

**Reply by sbf-** [reply]
Thanks for the extended logging...
As far as I can see, after receiving some (or sometimes a lot of) invalid packets, the right packet is received and in the end all comes good.
I really don't understand what's happening.

In the log there are also some Protocol-ID 0x6081 "Extended Energy Meter Protocol" packets. Maybe the two are related
Example of Unicast Extended Energy Meter Protocol packet:

```
32 bytes received from [192.168.2.7]
--------: 00 01 02 03 04 05 06 07 08 09
00000000: 53 4D 41 00 00 04 02 A0 00 00 
00000010: 00 01 00 0C 00 10 60 81 00 01 
00000020: 01 99 B3 54 A2 25 FF FF 00 00 
00000030: 00 00 
L2 header not found.
```

The 0x6081 @ pos 0x16 is the protocol-ID

So, for now I can't do anything

**Reply by tannegarn** [reply]
Thanks for looking into this. I do not have any energy power meter, just one SMA inverter. When reading the archiveDayData packages, the run is simply aborted and the CSV file is NOT created.

**Reply by sbf-** [reply]
> I do not have any energy power meter, just one SMA inverter

I know, the EM packet is coming from your inverter (IP192.168.2.7)

>When reading the archiveDayData packages, the run is simply aborted and the CSV file is NOT created.

Strange, because the logout procedure is executed every time, so the export should be OK (CSV_Export=1)

**Reply by tannegarn** [reply]
I will create a new log file with argument D5 and hopefully without archiveDayData.

**Comment by tannegarn** [comment]
[KP240520.txt](https://github.com/SBFspot/SBFspot/files/15374834/KP240520.txt)
[pvdiary_autorun_20240520.log](https://github.com/SBFspot/SBFspot/files/15374305/pvdiary_autorun_20240520.log)
Attached a logfile of this morning with 592 packet ID mismatches. Most are succesfull after a retry. The last 2 mismatches are for getting the archiveDayData and are not retried. this results in no data written to the CSV file.

	Regel 21593: Packet ID mismatch. Expected 19, received 22777
	Regel 22596: Packet ID mismatch. Expected 19, received 22790
	Regel 23379: Packet ID mismatch. Expected 18, received 22804
	Regel 25240: Packet ID mismatch. Expected 18, received 22830
Thanks for looking into this.

**Reply by sbf-** [reply]
Thanks, I've seen the issue now... I'll have a look

What is the `KP240520.txt` log file?

**Reply by tannegarn** [reply]
The KP240520 file is downloaded from the Sunny Boy data folder. Everyday, a KP and a HPyymmdd.txt file are created with a log of activities. I have tried to match anything obscure in these logfiles with packet ID mismatches. No match found.

**Comment by sbf-** [comment]
Need to fix #700
